azurelinuxagent/common/protocol/util.py

```diff
@@ -46,6 +46,13 @@
 WIRE_PROTOCOL_NAME = "WireProtocol"
 
 
+class NamedSet(set):
```

Comment:
We may want to rename this class. 'name' not being an item in the NamedSet raises an exception, which is not quite the behavior of a set in Python.

Reply:
This can be named as AttributeSet, where each member of the set actually becomes an attribute. I am open to other ideas as well.

Reply:
Renamed to AttributeEnum.